discovery/build.gradle

```diff
@@ -0,0 +1,17 @@
+dependencies {
+  implementation project(':util')
+  implementation project(':ethereum')
+  implementation project(':ethereum:datastructures')
```

Comment:
I think we can actually get rid of these dependencies.  `ethereum` and `ethereum:datastructures` already appear to be unused.  `util` is only referenced in two places - the discovery `Utils.leftPad` method's Javadoc (referencing `Bytes4` but actually only ever uses `Bytes`) and `CryptoUtil` to create a `MessageDigest`.

I'm somewhat tempted to make this a library in a separate repo as we need discv5 in besu as well.  In that case `CryptoUti.` could basically just inline what `BouncyCastleMessageDigestFactory` is doing to ensure it gets a bouncy castle sha256 implementation.

Reply:
Fixed leftPad, thanks for noticing it!